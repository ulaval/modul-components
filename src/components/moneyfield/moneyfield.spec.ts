import { mount, Wrapper } from '@vue/test-utils';
import Vue from 'vue';
import { renderComponent } from '../../../tests/helpers/render';
import uuid from '../../utils/uuid/uuid';
import MoneyfieldPlugin, { MMoneyfield } from './moneyfield';


jest.mock('../../utils/uuid/uuid');
(uuid.generate as jest.Mock).mockReturnValue('uuid');

describe('MTextfield', () => {
    beforeEach(() => {
        Vue.use(MoneyfieldPlugin);
    });

    it('should render correctly', () => {
        const component: Wrapper<MMoneyfield> = mount(MMoneyfield, {
            localVue: Vue
        });

        return expect(renderComponent(component.vm)).resolves.toMatchSnapshot();
    });
});
