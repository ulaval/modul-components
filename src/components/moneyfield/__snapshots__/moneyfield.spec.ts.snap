// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MTextfield should render correctly 1`] = `
<div><span>
        CA$
    </span>
  <div class="m-integerfield" style="width:100%;max-width:288px;">
    <div class="m-input-style m--is-tag-default" style="width:undefined;margin-top:10px;">
      <div class="m-input-style__body" style="top:0;">
        <div class="m-input-style__body__left"><input id="mIntegerfield-uuid" maxlength="16" pattern="[0-9]*" inputmode="numeric" type="number" value="" class="m-integerfield__input"> </div>
        <div class="m-input-style__body__right"></div>
      </div>
    </div>
    <div class="m-integerfield__validation">
      <div class="m-validation-message m-integerfield__validation__message"> </div>
    </div>
  </div>
</div>
`;
